File: python/백준/정렬/1026 보물.py
n = int(input())

array_a = list(map(int, input().split()))
array_b = list(map(int, input().split()))

array_a.sort()
array_b.sort(reverse=True)

sum = 0
for i in range(n):
    sum += array_b[i] * array_a[i] 

print(sum)

File: cpp/백준/정렬/1026 보물.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <functional>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<int> array_a(n), array_b(n);
    for (int i = 0; i < n; ++i) cin >> array_a[i];
    for (int i = 0; i < n; ++i) cin >> array_b[i];

    sort(array_a.begin(), array_a.end());
    sort(array_b.begin(), array_b.end(), greater<int>());

    long long sum = 0;
    for (int i = 0; i < n; ++i) {
        sum += (long long)array_b[i] * array_a[i];
    }

    cout << sum << "\n";
    return 0;
}
